analyses/cell-type-consensus/scripts/04-assign-consensus-celltypes.R

```diff
@@ -169,4 +200,54 @@ all_assignments_df <- celltype_df |>
   dplyr::mutate(consensus_annotation = dplyr::if_else(is.na(consensus_annotation) & (sample_type != "cell line"), "Unknown", consensus_annotation))
 
 # export file
-readr::write_tsv(all_assignments_df, opt$output_file)
+readr::write_tsv(all_assignments_df, opt$consensus_output_file)
+
+# Save gene expression file ----------------------------------------------------
+
+
+# list of all marker genes
+all_markers <- markers_df |> 
+  dplyr::pull(ensembl_gene_id) |> 
+  unique()
+
+# we only care about if that gene is expressed otherwise we won't waste memory and include it
+expressed_genes <- rowData(sce) |> 
+  as.data.frame() |>
+  dplyr::filter(detected > 0) |> 
+  dplyr::pull(gene_ids)
+
+# get markers that are expressed
+expressed_markers <- intersect(all_markers, expressed_genes)
+
+# if no markers are found, fill in columns with NA
+if(length(expressed_markers) == 0)(
+  
+  gene_exp_df <- data.frame(
+    barcodes = rownames(sce),
+    library_id = library_id, 
+    ensembl_gene_id = NA,
+    gene_expression = NA, 
+    validation_group_annotation = NA,
+    validation_group_ontology = NA
+  )
+  
+) else {
+  
+  # get logcounts from sce for expressed genes 
+  gene_exp_df <- scuttle::makePerCellDF(
+    sce, 
+    features = expressed_markers, 
+    assay.type = "logcounts",
+    use.dimred = FALSE
+  ) |>
+    tidyr::pivot_longer(starts_with("ENSG"), names_to = "ensembl_gene_id", values_to = "logcounts") |> 
+    # only pull out relevant columns
+    dplyr::select(barcodes, ensembl_gene_id, logcounts) |> 
+    # add library id column
+    dplyr::mutate(library_id = library_id) |> 
+    dplyr::relocate(library_id, .before = 0)
```

Comment:
Mutate has it's own `.before` arg.
```suggestion
    dplyr::mutate(library_id = library_id, .before = 0)
```